src/site-packages/openbmp/mrt2bmp/CollectorSender.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+"""OpenBMP mrt2bmp
+
+  Copyright (c) 2013-2015 Cisco Systems, Inc. and others.  All rights reserved.
+  This program and the accompanying materials are made available under the
+  terms of the Eclipse Public License v1.0 which accompanies this distribution,
+  and is available at http://www.eclipse.org/legal/epl-v10.html
+
+  .. moduleauthor:: Tim Evens <tievens@cisco.com>
+"""
+import socket
+import multiprocessing
+import time
+import Queue
+
+from time import sleep
+from openbmp.mrt2bmp.logger import init_mp_logger
+
+class BMPWriter(multiprocessing.Process):
+    """ BMP Writer
+
+        Pops messages from forwarder queue and transmits them to remote bmp collector.
+    """
+
+    def __init__(self, cfg, forward_queue, log_queue):
+        """ Constructor
+
+            :param cfg:             Configuration dictionary
+            :param forward_queue:   Output for BMP raw message forwarding
+            :param log_queue:       Logging queue - sync logging
+        """
+        multiprocessing.Process.__init__(self)
+        self._stop = multiprocessing.Event()
+
+        self.init_message = None
+        self.term_message = None
+        self.peer_up_messages = None
+
+        self._cfg = cfg
+        self._fwd_queue = forward_queue
+        self._log_queue = log_queue
+        self.LOG = None
+        self._isConnected = False
+        self._delay_after_peer_ups = self._cfg['collector']['delay_after_init_and_peer_ups']
+
+        self._sock = None
+
+    def run(self):
+        """ Override """
+        self.LOG = init_mp_logger("bmp_writer", self._log_queue)
+
+        self.LOG.info("Running bmp_writer")
+
+        self.connect()
+
+        try:
+            # Read queue
+            while not self.stopped():
+
+                # Do not pop any message unless connected
+                if self._isConnected:
+
+                    if not self._fwd_queue.empty():
```

Comment:
There's no sleep in the loop.  **When the queue is empty this loops continuously**, causing the CPU to stay at 50 - 80% for this process when idle.  You don't need to check if the queue is empty because you can use the ```get()``` method.  It will block if the queue is empty.  

This is what you want to do unless you plan to do something else while it's empty, which is not currently the case.